### DISCUSSION TITLE
PZEM-004-T and DS18x20 are not work simultaneously

### DISCUSSION THREAD

**Discussion by Csutiman** [arendst/Tasmota#22098] [Support request]
### PROBLEM DESCRIPTION
Using PZEM-004-T and DS18x20 separately is working, but when both are applied only PZEM-004-T is working.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): esp8266 12f
- [x] Tasmota binary firmware version number used: 14.2.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Tasmote web flasher
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
20:48:37.043 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
20:48:37.100 SRC: Backlog
20:48:37.103 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
20:48:37.109 MQT: stat/tasmota_754518/RESULT = {"NAME":"Spa Monitor","GPIO":[1,1504,1,1472,1312,1,1,1,0,1,1,1,1,1],"FLAG":0,"BASE":18}
20:48:37.315 SRC: Backlog
20:48:37.318 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
20:48:37.323 MQT: stat/tasmota_754518/RESULT = {"Module":{"18":"Generic"}}
20:48:37.516 DMP: A1 00 00 00 00 00 A1
20:48:37.565 SRC: Backlog
20:48:37.569 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
20:48:37.578 MQT: stat/tasmota_754518/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"1504":"PZEM004 Rx"},"GPIO2":{"0":"None"},"GPIO3":{"1472":"PZEM0XX Tx"},"GPIO4":{"1312":"DS18x201"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
20:49:21.726 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 19, Pld -99, Data 'Rule1; Rule2; Rule3'
20:49:21.832 SRC: Backlog
20:49:21.835 CMD: Grp 0, Cmd 'RULE', Idx 1, Len 0, Pld -99, Data ''
20:49:21.842 MQT: stat/tasmota_754518/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
20:49:21.847 DMP: A2 00 00 00 00 00 A2
20:49:22.059 SRC: Backlog
20:49:22.063 CMD: Grp 0, Cmd 'RULE', Idx 2, Len 0, Pld -99, Data ''
20:49:22.073 MQT: stat/tasmota_754518/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
20:49:22.311 SRC: Backlog
20:49:22.314 CMD: Grp 0, Cmd 'RULE', Idx 3, Len 0, Pld -99, Data ''
20:49:22.322 MQT: stat/tasmota_754518/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
20:50:11.142 MQT: stat/tasmota_754518/STATUS = {"Status":{"Module":18,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_754518","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
20:50:11.153 MQT: stat/tasmota_754518/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Power On","Uptime":"0T00:02:09","StartupUTC":"2024-09-04T19:48:02","Sleep":50,"CfgHolder":4617,"BootCount":78,"BCResetTime":"2024-08-18T21:45:21","SaveCount":157,"SaveAddress":"F8000"}}
20:50:11.162 MQT: stat/tasmota_754518/STATUS2 = {"StatusFWR":{"Version":"14.2.0(release-tasmota)","BuildDateTime":"2024-08-14T12:36:35","Boot":7,"Core":"2_7_7","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"433/699"}}
20:50:11.170 MQT: stat/tasmota_754518/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":4,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Telekom PbHtCY",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
20:50:11.184 MQT: stat/tasmota_754518/STATUS4 = {"StatusMEM":{"ProgramSize":648,"Free":352,"Heap":23,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,3,!4,!5,!6,7,!8,9,10,12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
20:50:11.195 MQT: stat/tasmota_754518/STATUS5 = {"StatusNET":{"Hostname":"tasmota-754518-1304","IPAddress":"192.168.1.130","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.10","DNSServer2":"8.8.8.8","Mac":"F4:CF:A2:75:45:18","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
20:50:11.203 MQT: stat/tasmota_754518/STATUS6 = {"StatusMQT":{"MqttHost":"homeassistant.local","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_754518","MqttUser":"homeassistant","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
20:50:11.213 MQT: stat/tasmota_754518/STATUS7 = {"StatusTIM":{"UTC":"2024-09-04T19:50:11Z","Local":"2024-09-04T20:50:11","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"06:12","Sunset":"19:25"}}
20:50:11.221 MQT: stat/tasmota_754518/STATUS9 = {"StatusPTH":{"PowerDelta":0,"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0,"MaxPower":0,"MaxPowerHold":10,"MaxPowerWindow":30,"MaxEnergy":0,"MaxEnergyStart":0}}
20:50:11.234 MQT: stat/tasmota_754518/STATUS10 = {"StatusSNS":{"Time":"2024-09-04T20:50:11","ENERGY":{"TotalStartTime":"2024-08-18T22:06:44","Total":29.913,"Yesterday":0.000,"Today":0.000,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":227,"Current":0.000}}}
20:50:11.244 MQT: stat/tasmota_754518/STATUS11 = {"StatusSTS":{"Time":"2024-09-04T20:50:11","Uptime":"0T00:02:09","UptimeSec":129,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Telekom PbHtCY","BSSId":"38:A6:59:EE:48:A6","Channel":1,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:05"}}}


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
20:50:14.503 NRG: EnergyTotal[0] 810.9130 kWh
20:50:14.748 DMP: A0 00 E3 01 00 00 84
20:50:15.506 DMP: A1 00 00 00 00 00 A1
20:50:15.754 DMP: A2 00 00 00 00 00 A2
20:50:16.519 DMP: A3 0C 5F A1 00 00 AF
20:50:16.521 NRG: EnergyTotal[0] 810.9130 kWh
20:50:16.766 DMP: A0 00 E3 00 00 00 83
20:50:17.522 DMP: A1 00 00 00 00 00 A1
20:50:17.767 DMP: A2 00 00 00 00 00 A2
21:15:04.767 NRG: EnergyTotal[0] 810.9130 kWh
21:15:05.073 DMP: A0 00 E3 07 00 00 8A
21:15:05.738 DMP: A1 00 00 00 00 00 A1
21:15:05.989 DMP: A2 00 00 00 00 00 A2
21:15:06.744 DMP: A3 0C 5F A1 00 00 AF
21:15:06.747 NRG: EnergyTotal[0] 810.9130 kWh

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
1. configure module type to generic
2. PZEM-004-T to GPIO1 (RX) and GPIO3 (TX)
3. press save, restart and check homepage
4. ![image](https://github.com/user-attachments/assets/803ca650-4caf-485b-9898-c707ed6ba783)
5. remove PZEM-004-T configs and configure GPIO 4 DS18x20
6. press save, restart and check homepage
7. ![image](https://github.com/user-attachments/assets/42251506-8b57-4ae3-8cc7-0eb5ccc2c8e4)
8. add to configuration of PZEM-004-T and keep DS18x20 on GPIO4
9.  press save, restart and check homepage
10. ![image](https://github.com/user-attachments/assets/177dffef-249b-4e99-916d-ec11dc0097f7)

### EXPECTED BEHAVIOUR
PZEM-004-T and DS18x20 can be used simultaneously.

### SCREENSHOTS
See in reproduction steps

### ADDITIONAL CONTEXT

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment]
What would be usefull are the logs at start when DS18x20 is supposed to be detected

If you configure both PZEM+DS18x20 but do not connect the PZEM (only DS18x20), can you see the DS18x20 ?

How is connected the DS18x20 ? Long line ? short line ? External pull-up ? Fully powered (not parasitic power) ?

**Reply by Csutiman** [reply]
What would be usefull are the logs at start when DS18x20 is supposed to be detected

- This is the startup log when PZEM is not configured

> 00:00:00.001 HDW: ESP8266EX
00:00:00.053 CFG: Loaded from flash at F4, Count 169
00:00:00.059 QPC: Count 1
00:00:00.061 CFG: CR 433/699, Busy 0
00:00:00.065 TYA: Active=0
00:00:00.069 ROT: Mode 1
00:00:00.070 SRC: Restart
00:00:00.071 Project tasmota - Tasmota Version 14.2.0(release-tasmota)-2_7_7(2024-08-14T12:36:35)
00:00:00.088 DSB: Sensors found 1
00:00:00.190 WIF: Checking connection...
00:00:00.191 WIF: Attempting connection...
00:00:01.001 WIF: Connecting to AP1 Telekom xxxxxx Channel 1 BSSId xx:xx:xx:xx:xx:xxin mode 11n as tasmota-754518-1304...
00:00:02.530 WIF: Checking connection...
00:00:02.531 WIF: Connected
00:00:02.782 HTP: Web server active on tasmota-754518-1304 with IP address 192.168.1.130
00:00:02.989 HTP: Main Menu
00:00:03.219 WIF: Sending Gratuitous ARP
00:00:03.220 NTP: Sync time...
00:00:03.247 WIF: DNS resolved '2.pool.ntp.org' (195.111.92.55) in 25 ms
00:00:03.260 RTC: UTC 2024-09-05T18:09:39Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
19:09:39.000 RTC: Synced by NTP
19:09:39.511 MQT: Attempting connection...
19:09:39.517 WIF: DNS resolved 'homeassistant.local' (192.168.1.10) in 4 ms
19:09:39.527 MQT: Connected
19:09:39.530 MQT: tele/tasmota_754518/LWT = Online (retained)
19:09:39.532 MQT: cmnd/tasmota_754518/POWER = 
19:09:39.533 MQT: Subscribe to cmnd/tasmota_754518/#
19:09:39.536 MQT: Subscribe to cmnd/tasmotas/#
19:09:39.538 MQT: Subscribe to cmnd/DVES_754518_fb/#
19:09:39.546 MQT: tele/tasmota_754518/INFO1 = {"Info1":{"Module":"Generic","Version":"14.2.0(release-tasmota)","FallbackTopic":"cmnd/DVES_754518_fb/","GroupTopic":"cmnd/tasmotas/"}}
19:09:39.550 MQT: tele/tasmota_754518/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-754518-1304","IPAddress":"192.168.1.130"}}
19:09:39.553 MQT: tele/tasmota_754518/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":85}}
19:09:40.761 HTP: Console
19:09:40.847 SRC: MQTT
19:09:40.849 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 1, Data '1'
19:09:40.855 MQT: stat/tasmota_754518/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:00:06","StartupUTC":"2024-09-05T18:09:34","Sleep":50,"CfgHolder":4617,"BootCount":84,"BCResetTime":"2024-08-18T21:45:21","SaveCount":169,"SaveAddress":"F4000"}}
19:09:40.947 SRC: MQTT
19:09:40.949 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 11, Data '11'
19:09:40.956 MQT: stat/tasmota_754518/STATUS11 = {"StatusSTS":{"Time":"2024-09-05T19:09:40","Uptime":"0T00:00:06","UptimeSec":6,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":33,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Telekom PbHtCY","BSSId":"38:A6:59:EE:48:A6","Channel":1,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:04"}}}
19:09:41.021 SRC: MQTT
19:09:41.024 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 10, Data '10'
19:09:41.030 MQT: stat/tasmota_754518/STATUS10 = {"StatusSNS":{"Time":"2024-09-05T19:09:41","DS18S20":{"Id":"0008026909CA","Temperature":24.4},"TempUnit":"C"}}
19:09:42.319 QPC: Reset
19:09:44.245 APP: Boot Count 85
19:09:44.252 MQT: tele/tasmota_754518/STATE = {"Time":"2024-09-05T19:09:44","Uptime":"0T00:00:10","UptimeSec":10,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Telekom PbHtCY","BSSId":"38:A6:59:EE:48:A6","Channel":1,"Mode":"11n","RSSI":60,"Signal":-70,"LinkCount":1,"Downtime":"0T00:00:04"}}
19:09:44.261 MQT: tele/tasmota_754518/SENSOR = {"Time":"2024-09-05T19:09:44","DS18S20":{"Id":"0008026909CA","Temperature":24.4},"TempUnit":"C"}
19:09:44.354 CFG: Saved to flash at FB, Count 170, Bytes 4096
19:09:49.280 MQT: tasmota/discovery/F4CFA2754518/config = {"ip":"192.168.1.130","dn":"Tasmota","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-754518-1304","mac":"F4CFA2754518","md":"Generic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"14.2.0","t":"tasmota_754518","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1} (retained)
19:09:49.290 MQT: tasmota/discovery/F4CFA2754518/sensors = {"sn":{"Time":"2024-09-05T19:09:49","DS18S20":{"Id":"0008026909CA","Temperature":24.4},"TempUnit":"C"},"ver":1} (retained)
19:09:50.511 SRC: MQTT
19:09:50.514 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 10, Data '10'
19:09:50.521 MQT: stat/tasmota_754518/STATUS10 = {"StatusSNS":{"Time":"2024-09-05T19:09:50","DS18S20":{"Id":"0008026909CA","Temperature":24.4},"TempUnit":"C"}}

If you configure both PZEM+DS18x20 but do not connect the PZEM (only DS18x20), can you see the DS18x20 ?

- No. But I can still see the PZEM status info, but it does not reduces to a single phase, it stay at three phases

How is connected the DS18x20 ? Long line ? short line ? 

- Short I would say, 20 cm.

External pull-up ? Fully powered (not parasitic power) ?

- External pull-up, 4k7 and fully powered. Double checked.
